apps/morph_validate

```diff
@@ -0,0 +1,291 @@
+
+'''Example of basic morphology validation'''
+from neurom.io.utils import load_neuron, get_morph_files
+from neurom.ezy import load_population
+from neurom.features import NEURITEFEATURES as nf
+from neurom import stats
+from neurom.exceptions import SomaError, IDSequenceError, MultipleTrees, MissingParentError
+from neurom.check import ok as chk_ok
+
+from collections import OrderedDict
+import argparse
+import functools
+import logging
+import yaml
+import os
+import sys
+import json
+
+
+DESCRIPTION = '''
+NeuroM Morphology Validation
+============================
+'''
+
+EPILOG = '''
+Description
+-----------
+
+Performs validation tests on reconstructed morphologies
+from data contained in morphology files, against a group
+of morphology files, that are defined as the target.
+
+Files are unloaded into neuron objects before testing.
+This means they must have a soma and no format errors.
+
+Default validation features
+------------------
+* Section number
+* Section lengths
+* Bifurcation angles
+* Section path distances
+* Section radial distances
+
+Default values for statistical test threshold
+--------------
+* Section number: 0.1
+* Section lengths: 0.1
+* Bifurcation angles: 0.1
+* Section path distances: 0.1
+* Section radial distances: 0.1
+
+Examples
+--------
+morph_check --help               # print this help
+morph_check some/path/neuron.h5  # Process an HDF5 file
+morph_check some/path/neuron.swc # Process an SWC file
+morph_check some/path/           # Process all HDF5 and SWC files found in directory
+'''
+
+L = logging.getLogger(__name__)
+
+
+def check_feature(feature, neuron, pop, fargs):
+    '''Runs a statistical test for a
+       selected feature and returns
+       True or False according to the results.
+    '''
+    neuron_data = nf[feature](neuron)
+    pop_data = nf[feature](pop)
+
+    results = stats.compare_two(neuron_data, pop_data)
+
+    return bool(results.dist < fargs)
+
+
+def _setup_logging(debug, log_file):
+    """ Set up logger """
+
+    fmt = logging.Formatter('%(levelname)06s: %(message)s')
+
+    level = logging.DEBUG if debug else logging.INFO
+    logging.basicConfig(level=level)
+    log = logging.getLogger()
+    log.handlers[0].setFormatter(fmt)
+
+    if log_file:
+        handler = logging.FileHandler(log_file)
+        handler.setFormatter(fmt)
+        log = logging.getLogger()
+        log.addHandler(handler)
+        log.setLevel(logging.DEBUG)
+
+
+def parse_args():
+    '''Parse command line arguments'''
+    parser = argparse.ArgumentParser(description=DESCRIPTION,
+                                     formatter_class=argparse.RawTextHelpFormatter,
+                                     epilog=EPILOG)
+    parser.add_argument('datapath',
+                        help='Path to morphology data file or directory')
+
+    parser.add_argument('targetpath',
+                        help='Path to population morphology data file or directory')
```

Comment:
This is going to be tricky, unless somebody puts the reference morphologies on the document service. Shouldn't we be comparing against some summary statistics values?